### PR TITLE
Add Oracle SQL `DROP DATABASE LINK`

### LINKED ISSUE #15954: [OpenForce 2022] Add Oracle SQL `DROP DATABASE LINK`
Hi community,

This issue is for [OpenForce 2022](https://github.com/apache/shardingsphere/discussions/12683#discussioncomment-2139441)

## Background
ShardingSphere parser engine helps users parse a SQL to get the AST (Abstract Syntax Tree) and visit this tree to get SQLStatement (Java Object). At present, this parser engine can handle SQLs for `MySQL`, `PostgreSQL`, `SQLServer`, `openGauss` and `Oracle`, which means we have to understand different database dialect SQLs.
More details: 
https://shardingsphere.apache.org/document/current/en/reference/sharding/parse/

## Task
This issue is to support parsing Oracle SQL `DROP DATABASE LINK`. You can learn more details about this SQL [here](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/DROP-DATABASE-LINK.html#GUID-89856C55-29FB-4B52-84A9-E53B8D115864).
- Remember to add new corresponding SQL cases in [SQL Cases](https://github.com/apache/shardingsphere/tree/master/shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported) and expected parsed results in [Expected Statment XML](https://github.com/apache/shardingsphere/tree/master/shardingsphere-test/shardingsphere-parser-test/src/main/resources/case).
- Run [SQLParserParameterizedTest](https://github.com/apache/shardingsphere/blob/master/shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/engine/SQLParserParameterizedTest.java) and [UnsupportedSQLParserParameterizedTest](https://github.com/apache/shardingsphere/blob/master/shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/engine/UnsupportedSQLParserParameterizedTest.java) to make sure no exceptions.

## Relevant Skills
1. Master JAVA language
2. Have a basic understanding of Antlr g4 file
3. Be familiar with Oracle SQLs

## Targets files
You can take this PR as a reference #14048 (Add Oracle SQL - CREATE DATABASE LINK)

### PR DESCRIPTION
**Pull request by everly-gif**
Fixes #15954

Changes proposed in this pull request:
- Adds `DROP DATABASE LINK`
- Added tests

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 [hunk 1]
```diff
@@ -2014,6 +2014,11 @@ createDatabaseLink
     : CREATE SHARED? PUBLIC? DATABASE LINK dbLink 
     (connectToClause | dbLinkAuthentication)* (USING connectString)?
     ;
+    
+dropDatabaseLink
+    : DROP PUBLIC? DATABASE LINK dbLink 
+    (connectToClause | dbLinkAuthentication)* (USING connectString)?
```
Comment (Liangda-w) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:2020]: This line is not part of `drop database link` grammar. See [doc](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/DROP-DATABASE-LINK.html#GUID-89856C55-29FB-4B52-84A9-E53B8D115864):

image

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/drop-database-link.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<sql-cases>
+    <sql-case id="drop_database_link" value="DROP PUBLIC DATABASE LINK remote USING 'remote'" db-types="Oracle" />
+    <sql-case id="drop_database_link_connect_to_user" value="DROP DATABASE LINK local CONNECT TO hr IDENTIFIED BY password USING 'local';" db-types="Oracle" />
+    <sql-case id="drop_database_link_connect_to_current_user" value="DROP DATABASE LINK remote.us.example.com CONNECT TO CURRENT_USER USING 'remote';" db-types="Oracle" />
```
Comment (Liangda-w) [inline shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/drop-database-link.xml:22]: Please correct these test cases according to the `drop database link` grammar

**Review by Liangda-w — CHANGES_REQUESTED** [review]
Hi @everly-gif, really nice PR! 
Please pay attention to the grammar, it's different from `create database link`.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 (hunk 1 above)
Comment (everly-gif) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4]: Referred the wrong docs :) However, I corrected it now. Do let me know if it's correct. On to the next requested changes!

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/drop-database-link.xml (hunk 2 above)
Comment (everly-gif) [reply shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/ddl/drop-database-link.xml]: Made the changes

**Review by Liangda-w — APPROVED** [review]
Looks good to me 👍 Thx for your great contribution!

**Comment by everly-gif** [comment]
Hi @Liangda-w  is there any more issues where I could add any other oracle SQL? I would love to work on some if any

**Comment by Liangda-w** [comment]
> Hi @Liangda-w is there any more issues where I could add any other oracle SQL? I would love to work on some if any

Sure! I'll find some Oracle SQL issues for you tomorrow 😄 glad that you enjoyed it!

**Pull request merged**